scipy/ndimage/tests/test_filters.py

```diff
@@ -178,7 +179,8 @@ def test_correlate01(self, xp):
         output = ndimage.convolve1d(array, weights)
         assert_array_almost_equal(output, expected)
 
-    @skip_xp_backends("jax.numpy", reasons=["output array is read-only."])
+    @skip_xp_backends("jax.numpy", reasons=["output array is read-only."],
+                      cpu_only=True, exceptions=['cupy', 'jax.numpy'])
```

Comment:
How can `jax.numpy` be both the first argument (the backend to skip) and in the `exceptions` list?

Reply:
Being the first argument means "always skip due to a reason in `reasons`". Being in `exceptions` means "don't get skipped by `cpu_only` when not on CPU as delegation is implemented".

In this case, I just went with the blanket "delegation is implemented for CuPy and JAX in this module", even though the support for JAX is patchy. In its current state, JAX could be removed from `exceptions` here. But if (somehow) JAX was no longer skipped via the first argument, we would want to add it back to `exceptions` to avoid the `cpu_only` skip.

Evidently this API is not perfectly intuitive, but I haven't thought of anything better yet. Please let me know if you do!

Reply:
Okay, that explanation seems reasonable. I can't think of something much cleaner. Explicitly explaining this case in the `skip_xp_backends` docstring would be useful, to avoid future confusion to the extent possible.